### PR TITLE
Add timeout auto increase until valid solution is found

### PR DESCRIPTION
**Pull request by Guillemdb**
Signed-off-by: Guillem Duran <guillem@sourced.tech>

### CODE REVIEW

File: seriate.py [hunk 1]
```diff
@@ -11,9 +11,47 @@
...
+        try:
+            route = _seriate(dists=dists, approximation_multiplier=approximation_multiplier,
+                             timeout=timeout)
+            break
+        except IncompleteSolutionError:
+            timeout *= 2
+    return route
+
+
+def _seriate(dists: numpy.ndarray, approximation_multiplier=1000, timeout=2.0) -> List[int]:
```
Comment (vmarkovtsev) [inline seriate.py:58]: Could you please change the docstring to indicate that the timeout is always greater than 0. Also `assert` on it.

File: seriate.py [hunk 2]
```diff
@@ -70,7 +108,12 @@ def dist_callback(x, y):
         search_parameters.time_limit.FromMilliseconds(int(timeout * 1000))
     search_parameters.local_search_metaheuristic = \
         routing_enums_pb2.LocalSearchMetaheuristic.GUIDED_LOCAL_SEARCH
+    search_parameters.first_solution_strategy = \
+        routing_enums_pb2.FirstSolutionStrategy.PATH_CHEAPEST_ARC
     assignment = routing.SolveWithParameters(search_parameters)
+    if assignment is None:
+        raise IncompleteSolutionError("No valid solution was found. Please increase the "
```
Comment (vmarkovtsev) [inline seriate.py]:
```suggestion
        raise IncompleteSolutionError("No solution was found. Please increase the "
```

File: seriate.py [hunk 3]
```diff
@@ -11,9 +11,47 @@
...
+    if timeout > 0:
+        return _seriate(dists=dists, approximation_multiplier=approximation_multiplier,
+                        timeout=timeout)
+    timeout = 1.
+    route = None
+    while route is None:
+        try:
+            route = _seriate(dists=dists, approximation_multiplier=approximation_multiplier,
+                             timeout=timeout)
+            break
```
Comment (vmarkovtsev) [inline seriate.py]:
```suggestion
```

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

**Comment by vmarkovtsev** [comment]
The test is missing. You need to use the file provided in the issue to raise the incomplete solution exception.

File: seriate.py (hunk 1 above)
Comment (vmarkovtsev) [reply seriate.py:58]: @Guillemdb This still holds.

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: seriate.py (hunk 1 above)
Comment (Guillemdb) [reply seriate.py:58]: `assert` should never be used to validate use input, as it meant to be used as a debugging tool that can be disabled in production. A ValueError should be raised instead to be sure that even disabling the asserts it cannot run with a timeout less than zero.

EDIT: dyslexiaaa!

**Review by vmarkovtsev — APPROVED** [review]

**Pull request merged**